OpenRA.Mods.Common/Traits/GivesCashOnCapture.cs

```diff
@@ -0,0 +1,57 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2016 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Mods.Common.Effects;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Lets the actor grant cash when captured.")]
+	class GivesCashOnCaptureInfo : UpgradableTraitInfo
+	{
+		[Desc("Whether to show the cash tick indicators rising from the actor.")]
+		public readonly bool ShowTicks = true;
+
+		[Desc("Amount of money awarded for capturing the actor.")]
+		public readonly int CaptureAmount = 0;
+
+		public override object Create(ActorInitializer init) { return new GivesCashOnCapture(this); }
+	}
+
+	class GivesCashOnCapture : UpgradableTrait<GivesCashOnCaptureInfo>, INotifyCapture
+	{
+		readonly GivesCashOnCaptureInfo info;
+
+		public GivesCashOnCapture(GivesCashOnCaptureInfo info)
+			: base(info)
+		{
+			this.info = info;
+		}
+
+		void INotifyCapture.OnCapture(Actor self, Actor captor, Player oldOwner, Player newOwner)
+		{
+			if (IsTraitDisabled)
+				return;
+
+			if (info.CaptureAmount > 0)
```

Comment:
Actually, I think `> 0` shouldn't ben enforced at all, neither in code nor via lint error, because some modders might want to use negative values for minigames and such. A mere lint warning should be enough.